src/domain/community/profile/profile.service.ts

```diff
@@ -41,21 +41,29 @@ export class ProfileService {
   async createProfile(profileData?: CreateProfileInput): Promise<IProfile> {
     const profile: IProfile = Profile.create({
       description: profileData?.description,
-      tagsets: profileData?.tagsetsData,
-      references: profileData?.referencesData,
     });
     profile.authorization = new AuthorizationPolicy();
     profile.avatar = await this.visualService.createVisualAvatar();
     profile.location = await this.locationService.createLocation(
       profileData?.location
     );
 
-    if (!profile.references) {
-      profile.references = [];
+    profile.references = [];
+    if (profileData?.referencesData) {
```

Comment:
With this code, if you put a `reference` with value, e.g. `https://alkem.io` and you already have a `reference` with value `https://alkem.io` you will end up having semantically the same reference twice. It is the same for `tags` in a `tagset`. I don't believe this is desired behavior.

Reply:
My bad, there is a check in the reference service